diff --git a/package/boot/uboot-rockchip/Makefile b/package/boot/uboot-rockchip/Makefile
index 1633d1ef57..1275a3f0d7 100644
--- a/package/boot/uboot-rockchip/Makefile
+++ b/package/boot/uboot-rockchip/Makefile
@@ -191,6 +191,13 @@ define U-Boot/rk3528/Default
   TPL:=$(RK3528_TPL)
 endef
 
+define U-Boot/generic-rk3528
+  $(U-Boot/rk3528/Default)
+  NAME:=GENERIC RK3528
+  BUILD_DEVICES:= \
+    hinlink_opc-h28k
+endef
+
 define U-Boot/mangopi-m28c-rk3528
   $(U-Boot/rk3528/Default)
   NAME:=MangoPi M28C
@@ -537,6 +544,7 @@ UBOOT_TARGETS := \
   roc-cc-rk3328 \
   rock64-rk3328 \
   rock-pi-e-rk3328 \
+  generic-rk3528 \
   mangopi-m28c-rk3528 \
   mangopi-m28k-rk3528 \
   radxa-e20c-rk3528 \
