diff --git a/target/linux/rockchip/armv8/base-files/etc/board.d/01_leds b/target/linux/rockchip/armv8/base-files/etc/board.d/01_leds
index 504164d9d8..b4f046a8b0 100644
--- a/target/linux/rockchip/armv8/base-files/etc/board.d/01_leds
+++ b/target/linux/rockchip/armv8/base-files/etc/board.d/01_leds
@@ -50,6 +50,10 @@ widora,mangopi-m28k)
 	ucidef_set_led_netdev "wan" "WAN" "white:wan" "eth1"
 	ucidef_set_led_netdev "lan" "LAN" "white:lan" "eth0"
 	;;
+hinlink,opc-h28k)
+	ucidef_set_led_netdev "wan" "WAN" "blue:wan" "eth1"
+	ucidef_set_led_netdev "lan" "LAN" "amber:lan" "eth0"
+	;;
 esac
 
 board_config_flush
diff --git a/target/linux/rockchip/armv8/base-files/etc/board.d/02_network b/target/linux/rockchip/armv8/base-files/etc/board.d/02_network
index 4ef39acc04..988143a6c5 100644
--- a/target/linux/rockchip/armv8/base-files/etc/board.d/02_network
+++ b/target/linux/rockchip/armv8/base-files/etc/board.d/02_network
@@ -30,6 +30,7 @@ rockchip_setup_interfaces()
 	armsom,sige3|\
 	friendlyarm,nanopi-r5c|\
 	friendlyarm,nanopi-r76s|\
+	hinlink,opc-h28k|\
 	lunzn,fastrhino-r66s|\
 	mmbox,anas3035|\
 	nlnet,xiguapi-v3|\
